internal/xds/translator/listener.go

```diff
@@ -785,3 +790,55 @@ func buildHeadersWithUnderscoresAction(in *ir.HeaderSettings) corev3.HttpProtoco
 	}
 	return corev3.HttpProtocolOptions_REJECT_REQUEST
 }
+
+func buildForwardClientCertDetailsAction(in *ir.HeaderSettings) hcmv3.HttpConnectionManager_ForwardClientCertDetails {
+	if in != nil {
+		if in.ForwardClientCert != nil {
+			switch in.ForwardClientCert.Mode {
+			case ir.ForwardModeSanitize:
+				return hcmv3.HttpConnectionManager_SANITIZE
+			case ir.ForwardModeForwardOnly:
+				return hcmv3.HttpConnectionManager_FORWARD_ONLY
+			case ir.ForwardModeAppendForward:
+				return hcmv3.HttpConnectionManager_APPEND_FORWARD
+			case ir.ForwardModeSanitizeSet:
+				return hcmv3.HttpConnectionManager_SANITIZE_SET
+			case ir.ForwardModeAlwaysForwardOnly:
+				return hcmv3.HttpConnectionManager_ALWAYS_FORWARD_ONLY
+			}
+		}
+	}
+	return hcmv3.HttpConnectionManager_SANITIZE
+}
+
+func buildSetCurrentClientCertDetails(in *ir.HeaderSettings) *hcmv3.HttpConnectionManager_SetCurrentClientCertDetails {
+	if in == nil {
+		return nil
+	}
+
+	if in.ForwardClientCert == nil {
+		return nil
+	}
+
+	if len(in.ForwardClientCert.CertDetailsToAdd) == 0 {
+		return nil
+	}
```

Comment:
ditto

Reply:
I don't think we can merge this, there is possibility that headers & forwardclientcert is is nil.